lib/API/OAuth2/Repositories/ClientRepository.php

```diff
@@ -60,7 +61,7 @@ public function getClientEntity($clientIdentifier, $grantType, $clientSecret = n
         }
 
         if($client->allowed_grant_type !== $grantType) {
-            throw OAuthServerException::grantTypeNotAllowedForClient();
+            throw new GrantTypeNotAllowedForClient();
```

Comment:
In this case, GrantTypeNotAllowedForClient looks completely unnecessary to me...